packages/solid-query/src/createBaseQuery.ts

```diff
@@ -31,7 +31,7 @@ export function createBaseQuery<
   Observer: typeof QueryObserver,
 ): QueryObserverResult<TData, TError> {
   const queryClient = useQueryClient({ context: options.context })
-
+  const emptyData = Symbol('empty')
```

Comment:
would it be safer if we crated the symbol outside of `createBaseQuery` ? creating the symbol twice will not give referential identity. Just wondering if there is a case where `createBaseQuery` would run again and the comparison would fail.

I think because it's a const, when the resource functions runs it will closure over the emptyData from when it was created, but still: shouldn't there be _exactly one_ `emptyData` symbol ?

Reply:
That's a great question 😁 Solid JS hooks and components are created only once. Unlike react, They never 're-render'. The options and functions are made reactive using reactive utility functions that SolidJS provides like 'createComputed', 'createEffect', etc. The emptyData const will always have the same referential identity since it will only be created once, throughout the hook's lifespan.

Reply:
of course 🤦‍♂️. I'm still thinking in react re-renderings lol.